WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/OrderTotalsCalculator.swift

```diff
@@ -49,6 +49,15 @@ final class OrderTotalsCalculator {
             .reduce(NSDecimalNumber(value: 0), { $0.adding($1) })
     }
 
+    /// Total value of all coupons on an order.
+    ///
+    var couponsTotal: NSDecimalNumber {
```

Comment:
We could make this private and test the method through the `OrderTotalsCalculator`

```suggestion
    private var couponsTotal: NSDecimalNumber {
```

Then confirm that is working with coupons (these tests may need some re-work 😅 but should cover the cases for empty coupon and one coupon ):

```
    func test_updateOrderTotal_when_has_no_coupons_then_returns_order_with_expected_total() {
        // Given
        let currencyFormatter = CurrencyFormatter(currencySettings: CurrencySettings())
        let order = Order.fake().copy(shippingTotal: "5.00",
                                      totalTax: "3.00",
                                      items: [OrderItem.fake().copy(subtotal: "2.00"), OrderItem.fake().copy(subtotal: "8.00")],
                                      coupons: [],
                                      fees: [OrderFeeLine.fake().copy(total: "2.00"), OrderFeeLine.fake().copy(total: "8.00")])

        // When
        let orderTotalsCalculator = OrderTotalsCalculator(for: order, using: currencyFormatter)
        let updatedOrder = orderTotalsCalculator.updateOrderTotal()

        // Then
        XCTAssertEqual(updatedOrder.total, "28")
    }

    func test_updateOrderTotal_when_has_coupons_then_returns_order_with_expected_total() {
        // Given
        let coupons = OrderCouponLine.fake().copy(discount: "10")
        let currencyFormatter = CurrencyFormatter(currencySettings: CurrencySettings())
        let order = Order.fake().copy(shippingTotal: "5.00",
                                      totalTax: "3.00",
                                      items: [OrderItem.fake().copy(subtotal: "2.00"), OrderItem.fake().copy(subtotal: "8.00")],
                                      coupons: [coupons],
                                      fees: [OrderFeeLine.fake().copy(total: "2.00"), OrderFeeLine.fake().copy(total: "8.00")])

        // When
        let orderTotalsCalculator = OrderTotalsCalculator(for: order, using: currencyFormatter)
        let updatedOrder = orderTotalsCalculator.updateOrderTotal()

        // Then
        XCTAssertEqual(updatedOrder.total, "18")
    }
```

Reply:
Thanks for the comment here. I followed the pattern of that class, where all the other properties were done that way.